connectors-e2e-test/connectors-e2e-test-message/src/test/java/io/camunda/connector/e2e/MessageTests.java

```diff
@@ -0,0 +1,221 @@
+@ComponentScan(basePackages = {"io.camunda.connector.runtime.app"})
+public class MessageTests {
+
+  private static final String MESSAGE_NAME = "first-message";
+  private static final String CORRELATION_KEY_EXPRESSION = "valOneCorrelation";
+  private static final String CORRELATION_VALUE = "123";
+
+  @TempDir File tempDir;
+
+  @Autowired CamundaClient camundaClient;
+  private ProcessInstanceEvent receiveInstance;
+  private BpmnModelInstance publishMessageModel;
+
+  @BeforeEach
+  public void createReceiveInstance() {
+    camundaClient
+        .newDeployResourceCommand()
+        .addProcessModel(
+            Bpmn.createExecutableProcess("receiveProcess")
+                .startEvent()
+                .intermediateCatchEvent()
+                .message(
+                    m ->
+                        m.name(MESSAGE_NAME)
+                            .zeebeCorrelationKeyExpression(CORRELATION_KEY_EXPRESSION))
+                .endEvent()
+                .done(),
+            "receiveMessage.bpmn")
+        .send()
+        .join();
+
+    receiveInstance =
+        camundaClient
+            .newCreateInstanceCommand()
+            .bpmnProcessId("receiveProcess")
+            .latestVersion()
+            .variable(CORRELATION_KEY_EXPRESSION, CORRELATION_VALUE)
+            .send()
+            .join();
+
+    publishMessageModel = createSendTaskModel();
+  }
+
+  @Test
+  public void testPublishMessage() {
+    var sendTaskTemplate =
+        createSendTaskTemplate(
+            "publish", "={messageKeyResult: messageKey, tenantIdResult: tenantId}");
+
+    BpmnModelInstance messageModelWithElTemplate =
+        new BpmnFile(publishMessageModel)
+            .writeToFile(new File(tempDir, "message_send.bpmn"))
+            .apply(
+                sendTaskTemplate,
+                "sendTask",
+                new File(tempDir, "message_send_with_el_template.bpmn"));
+
+    ProcessInstanceEvent processInstance =
+        ZeebeTest.with(camundaClient)
+            .deploy(messageModelWithElTemplate)
+            .createInstance()
+            .waitForProcessCompletion()
+            .getProcessInstanceEvent();
+
+    assertThat(processInstance)
+        .hasVariableNames("messageResponse", "messageKeyResult", "tenantIdResult")
+        .hasVariable("tenantIdResult", "<default>");
+
+    assertThatReceiveInstanceIsEnded();
+  }
+
+  @Test
+  @Disabled(
+      """
+Unauthorized access to correlate message REST API:
+Details from surefire report:
+2025-03-03T08:45:02.658+01:00 DEBUG 99400 --- [pool-7-thread-1] i.c.c.r.c.outbound.ConnectorJobHandler   : Exception while processing job: 2251799813685359 for tenant: <default>
+
+io.camunda.zeebe.client.api.command.ProblemException: Failed with code 401: 'Unauthorized'. Details: 'class ProblemDetail {
+    type: about:blank
+    title: Unexpected server response
+    status: 500
+    detail: {"message":"Full authentication is required to access this resource"}
+    instance: null
+}'
```

Comment:
interesting finding, do you already have an idea if this is a problem in `CamundaProcessTest`? We can follow up with the CPT team to clarify

Reply:
I don't have an idea yet. I hope that both sides: client implementation and server authorization, may change with further development and future alpha releases. 

It's worth to give it another try.